### PR TITLE
Align with latest template-multi project

### PR DESCRIPTION
**Pull request by pderop**
Motivation:

The template-multi project has evolved since the codec-multipart was created from it, so, we need to align this project with the latest template-multi, like the .github workflow, etc...
All this is necessary in preparation for a release.

While aligning this project to the template-multi, I also found that the template contains a "leak" profile with this:

```
        <test.argLine>-Dio.netty.leakDetectionLevel=paranoid -Dio.netty.leakDetection.targetRecords=32</test.argLine>
```

I think it's wrong, and the following properties should be used instead of the above (when the tests must be run with leak detector):

```
+        <test.argLine>-Dio.netty5.leakDetection.level=paranoid -Dio.netty5.leakDetection.targetRecords=32 -Dio.netty5.buffer.leakDetectionEnabled=true -Dio.netty5.buffer.lifecycleTracingEnabled=true</test.argLine>
```

So, this PR is now using the correct leak detection properties (I will make another PR in order to fix this in the template-multi).

Motifications:

Aligned the .github/ dir, the parent pom.xml and the SECURITY.md with the one from the template-multi.
Also did minor fixes in README.adoc

Results:

the project structure is closer to the other netty-contrib projects and is in a releasable state.
Also, the tests can now be run with the leak detector using the "leak" profile.

### CODE REVIEW

**Comment by pderop** [comment]
There are some leaks ! so the CI build fail ... I'll investigate with probably another PR.

**Comment by pderop** [comment]
Rebased on top of main to pick up #15

**Comment by pderop** [comment]
rebased on top of main to pick up #17

**Comment by pderop** [comment]
@chrisvest ,

Can you take a look ?

It's just an alignment of the MP project with other netty-contrib project structures (.github files, etc ...), so it will bring the project closer to a releasable state (hopefully :sweat_smile:)

thanks.

File: pom.xml [hunk 1]
```diff
@@ -67,7 +67,7 @@
 
   <properties>
     <netty.version>5.0.0.Alpha5</netty.version>
-    <netty.build.version>30</netty.build.version>
+    <netty.build.version>29</netty.build.version>
```
Comment (chrisvest) [inline pom.xml]: Downgrading? Do you have plans for a round of dependency upgrades later?

**Review by chrisvest — APPROVED** [review]
Hey, finally took a look at this.

File: pom.xml (hunk 1 above)
Comment (pderop) [reply pom.xml]: thanks for reviewing;

so, I have aligned with other projects as close as possible; and all other netty-contrib projects are using netty.build.version=29, so, I then downgraded from 30 to 29 (initially, when I created the project, I used netty.build.version=30, but honestly, I don't remember why :thinking:)

I see that latest version of io.netty:netty-build-common is 31, so maybe better is to use the latest version right ?

Comment (chrisvest) [reply pom.xml]: Yeah. I don't know what's changed, but I assume newer is better. We can make the changes one at a time, just in case something breaks.

**Comment by pderop** [comment]
thanks for the review,

I got some feedback from my team, so maybe I'll do another PR (hopefully the very last one before releasing :rocket:)

**Pull request merged**